pkg/cloudprovider/azure.go

```diff
@@ -658,3 +691,59 @@ func (a *Azure) getNodeLock(nodeName string) *sync.Mutex {
 func getNameFromResourceID(id string) string {
 	return id[strings.LastIndex(id, "/"):]
 }
+
+// watchForChanges closes stopCh if the configuration file changed.
```

Comment:
NIT: this function is not specific to config files, can you update the comment to better reflect what it does?

Reply:
Yeah I'll make it more generic